src/Acuminator/Acuminator.Analyzers/StaticAnalysis/MethodsUsageInDac/MethodsUsageInDacAnalyzer.cs

```diff
@@ -46,29 +50,53 @@ private void AnalyzeDacDeclaration(SyntaxNodeAnalysisContext syntaxContext, PXCo
 					}
 					else if (node is PropertyDeclarationSyntax property)
 					{
-						AnalyzeMethodInvocationInDacProperty(property, syntaxContext, pxContext);
+						AnalyzeMethodInvocationInDacProperty(property, whiteList, syntaxContext, pxContext);
 					}
 				}
 			}
 		}
 
-		private void AnalyzeMethodInvocationInDacProperty(PropertyDeclarationSyntax property,
-			SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        private IEnumerable<INamedTypeSymbol> GetWhitelist(PXContext pxContext)
+        {
+            return new[]
+            {
+                pxContext.SystemTypes.Bool,
```

Comment:
@metallmonster meybe we should add enums and float here

Reply:
@SENya1990 I looked into our code, we have PXFloatAttribute, so I'll add float
Enums aren't analyzed by this diagnostic

Reply:
@metallmonster float should be added just in case - it's strange to not support one of the primitive types. 
Will the diagnostic show error on this code:
```
public enum MyEnum { Val };

public class DAC : IBqlTable
{
    public string Property 
    {
       get
       {
           return Val.ToString();
        }
     }
}
```

Reply:
@SENya1990 yes, it will. Now it is fixed.